battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java

```diff
@@ -27,11 +27,12 @@
 
     @Override
     public void run() throws IOException {
-        String fromServer;
-        while (!("Provide ships").equals(fromServer = in.readLine())) {
+        String fromServer = in.readLine();
+        while (!("Provide ships").equals(fromServer)) {
             if (LOGGER.isLoggable(Level.INFO)) {
                 LOGGER.info("Server: " + fromServer);
             }
+            fromServer = in.readLine();
```

Comment:
So, before the while, you are setting `fromServer` to whatever comes from server, in hopes in is `Provide ships`. As long as it is, you are changing it? What should `fromServer` have here, after this substitution is done?

Reply:
We are looking for "Provide ships" message which is last of initial messages. All other initial messages are logged. So the answer for your question is: Server should have "Provide ships".

Reply:
I dislike the fact that messages are here treated as a String. If they change, you guys are blindsided to it. Have they been types or enums or enum fields, that would've made you safer and this code - more extendable. Feel free to log this as an issue for later and proceed.

Reply:
Thanks for the explanation.